pkg/path/extract_test.go

```diff
@@ -32,6 +46,107 @@ func TestMain(m *testing.M) {
 }
 
 func Test_ExtractPaths(t *testing.T) {
+	assert.Equal(t, 2, len(roPaths))
+	for _, roPath := range roPaths {
+		switch path := roPath.Path; path {
+		case "/t1:cont1a/t1:cont2a/t1:leaf2c":
+			assert.Equal(t, 1, len(roPath.SubPath))
+			sp := roPath.SubPath[0]
+			assert.Equal(t, "leaf2c", sp.AttrName)
+		case "/t1:cont1b-state":
+			assert.Equal(t, 3, len(roPath.SubPath))
+			for _, sp := range roPath.SubPath {
+				switch subPath := sp.SubPath; subPath {
+				case "/t1:leaf2d":
+					assert.Equal(t, "leaf2d", sp.AttrName)
+				case "/t1:list2b[t1:index=*]/t1:index":
+					assert.Equal(t, "index", sp.AttrName)
+				case "/t1:list2b[t1:index=*]/t1:leaf3c":
+					assert.Equal(t, "leaf3c", sp.AttrName)
+				default:
+					t.Fatalf("unexpected subpath %s for RO path %s", subPath, path)
+				}
+			}
+		default:
+			t.Fatalf("unexpected RO path %s", path)
+		}
+	}
+
+	assert.Equal(t, 21, len(rwPaths))
+	for _, rwPath := range rwPaths {
+		switch path := rwPath.Path; path {
+		case "/t1:leafAtTopLevel":
+			assert.Equal(t, "leafAtTopLevel", rwPath.AttrName)
+		case "/t1:cont1a/t1:cont2a/t1:leaf2a":
+			assert.Equal(t, "leaf2a", rwPath.AttrName)
+		case "/t1:cont1a/t1:cont2a/t1:leaf2b":
+			assert.Equal(t, "leaf2b", rwPath.AttrName)
+		case "/t1:cont1a/t1:cont2a/t1:leaf2d":
+			assert.Equal(t, "leaf2d", rwPath.AttrName)
+		case "/t1:cont1a/t1:cont2a/t1:leaf2e":
+			assert.Equal(t, "leaf2e", rwPath.AttrName)
+		case "/t1:cont1a/t1:cont2a/t1:leaf2f":
+			assert.Equal(t, "leaf2f", rwPath.AttrName)
+		case "/t1:cont1a/t1:cont2a/t1:leaf2g":
+			assert.Equal(t, "leaf2g", rwPath.AttrName)
+		case "/t1:cont1a/t1:leaf1a":
+			assert.Equal(t, "leaf1a", rwPath.AttrName)
+		case "/t1:cont1a/t1e:list4[t1e:id=*]/t1e:id":
```

Comment:
@adibrastegarnia note the `t1e` on `list4` - this is because `list4` comes from the augmented `extra` file.
Prefixes are not always the same all the way down.